src/gui/N2Go_Gui.php

```diff
@@ -251,18 +255,17 @@ private function refreshTokens()
             )
         );
 
-        if (is_wp_error($responseRaw)) {
-            $this->apiErrorMessage = $responseRaw->get_error_message($responseRaw->get_error_code());
-        } else {
+        if($this->verifyResponse($responseRaw)){
             $response = json_decode($responseRaw['body']);
 
-            if (isset($response->access_token) && !empty($response->access_token)) {
+            if (isset($response->access_token) && !empty($response->access_token) && isset($response->refresh_token) && !empty($response->refresh_token)) {
```

Comment:
just to make the code more readable. both isset()s could be removed since empty() already checks if the values are set and dose not throws an error if not